src/test/java/org/kie/processmigration/service/SchedulerServiceImplTest.java

```diff
@@ -92,9 +90,13 @@ void testScheduleMigrationNow() throws InvalidMigrationException, InterruptedExc
         schedulerService.scheduleMigration(migration);
 
         // Then
-        count.await(10, TimeUnit.SECONDS);
+        if (!count.await(10, TimeUnit.SECONDS)) {
+            fail("Failed while waiting for the jobs to be completed");
+        }
         verify(migrationService, times(1)).get(99L);
         verify(migrationService, times(1)).migrate(migration);
+
+        Thread.sleep(1000L); // wait for the triggers/jobs to be deleted from the db
```

Comment:
Can't we avoid using `Thread.sleep()` ?

Reply:
removing last checks, since they are not part of the goal of these tests. Also, seems there is some problems with the listeners and the checkExists method retrieve the job even though the job has been already completed.